src/Parquet/ParquetConvert.cs

```diff
@@ -97,13 +97,18 @@ public static class ParquetConvert {
         /// 
         /// </summary>
         /// <typeparam name="T"></typeparam>
-        /// <param name="input"></param>
-        /// <param name="rowGroupIndex"></param>
+        /// <param name="input">Stream</param>
+        /// <param name="rowGroupIndex">int</param>
+        /// <param name="fileSchema">Schema</param>
         /// <returns></returns>
-        public static async Task<T[]> DeserializeAsync<T>(Stream input, int rowGroupIndex = -1) where T : new() {
+        public static async Task<T[]> DeserializeAsync<T>(Stream input, int rowGroupIndex = -1, 
+            Schema fileSchema = null
+        ) where T : new() {
             var result = new List<T>();
             using(ParquetReader reader = await ParquetReader.CreateAsync(input)) {
-                Schema fileSchema = new SchemaReflector(typeof(T)).Reflect();
+                if (fileSchema == null) {
+                    fileSchema = new SchemaReflector(typeof(T)).Reflect();
```

Comment:
```suggestion
                    fileSchema = fileSchema ?? SchemaReflector.Reflect<T>();
```
pseudo code, but do this instead of if statement?

Reply:
also, should we be using the static methods over the public constructor, or is there a reason we don't want to cache this schema?  see my comment [here](https://github.com/aloneguid/parquet-dotnet/pull/202#discussion_r1038295559)